state/protocol/badger/mutator.go

```diff
@@ -809,49 +920,52 @@ SealLoop:
 				// validate the service event
 				err := isValidExtendingEpochSetup(ev, activeSetup, epochStatus)
 				if protocol.IsInvalidServiceEventError(err) {
-					// EECC - we have observed an invalid service event, which is
-					// an unrecoverable failure. Flag this in the DB and exit
-					ops = append(ops, transaction.WithTx(operation.SetEpochEmergencyFallbackTriggered(blockID)))
+					// we have observed an invalid service event, which triggers epoch fallback mode
+					insertingCandidateTriggersEpochFallback = true
```

Comment:
Rather than returning a database operation function regarding epoch fallback being triggered, now return a boolean (aiming to mimic the pattern of `epochFallbackTriggeredByFinalizedBlock`) and let the upper layer deal with the details of events/metrics/state changes associated with this.